### PR TITLE
build: add intel openmp support

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR adds a build flag to support building with Intel OpenMP.

### CODE REVIEW

File: docs/how_to_use/how_to_build.md [hunk 1]
```diff
@@ -266,3 +266,17 @@ unzip -o bazel-bin/docs/doxygen/tachyon_api_docs.zip &&
 google-chrome bazel-bin/docs/doxygen/html/index.html
 # generate HTML files and open on Chrome browser.
 ```
+
+## Performance Tuning
+
+### Use Intel OpenMP Runtime Library(libiomp)
+
+By default, Tachyon uses GNU OpenMP (GNU `libgomp`) for parallel computation. On Intel platforms, Intel OpenMP Runtime Library (`libiomp`) provides OpenMP API specification support. It sometimes brings more performance benefits compared to `libgomp`.
```
Comment (TomTaehoonKim) [inline docs/how_to_use/how_to_build.md:274]: Can you add the pytorch reference here for better credibility? https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#intel-openmp-runtime-library-libiomp

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: docs/how_to_use/how_to_build.md (hunk 1 above)
Comment (chokobole) [reply docs/how_to_use/how_to_build.md:274]: Why do we need this?

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Pull request merged**